File: Messengerv2/app/src/main/java/com/magnet/magntetchatapp/ui/activities/section/splash/SplashActivity.java
package com.magnet.magntetchatapp.ui.activities.section.splash;

import com.magnet.chatsdkcover.mvp.api.abs.SplashContract;
import com.magnet.chatsdkcover.mvp.presenters.DefaultSplashPresenter;
import com.magnet.magntetchatapp.R;
import com.magnet.magntetchatapp.ui.activities.abs.BaseActivity;
import com.magnet.magntetchatapp.ui.activities.section.home.HomeActivity;
import com.magnet.magntetchatapp.ui.activities.section.login.LoginActivity;

public class SplashActivity extends BaseActivity implements SplashContract.View {

    private SplashContract.Presenter presenter;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_splash;
    }

    @Override
    protected int getMenuId() {
        return NONE_MENU;
    }

    @Override
    protected void onCreateActivity() {
        presenter = new DefaultSplashPresenter(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        runOnMainThread(2, splashAction);
    }

    /**
     * Method which provide to performing of the navigation
     *
     * @param navigationType navigation type
     */
    @Override
    public void navigate(SplashContract.NavigationType navigationType) {
        switch (navigationType) {
            case HOME:
                startActivityWithClearTop(HomeActivity.class);
                break;
            case LOGIN:
                startActivityWithClearTop(LoginActivity.class);
                break;
            default:
                break;
        }

    }

    /**
     * Callback which provide the action after delay
     */
    private final OnActionPerformer splashAction = new OnActionPerformer() {
        @Override
        public void onActionPerform() {
            if (presenter != null) {
                presenter.onSplashAction();
            }
        }
    };


}

File: KMessenger/app/src/main/kotlin/com/magnet/kmessenger/ui/SplashActivity.kt
package com.magnet.kmessenger.ui

import android.os.Bundle
import com.magnet.kmessenger.R
import com.magnet.magnetchat.presenters.SplashContract
import com.magnet.magnetchat.presenters.impl.DefaultSplashPresenter
import com.magnet.magnetchat.ui.activities.MMXBaseActivity

/**
 * Created by aorehov on 26.05.16.
 */
class SplashActivity : MMXBaseActivity(), SplashContract.View {

    val presenter: SplashContract.Presenter by lazy { DefaultSplashPresenter(this) }

    override fun getLayoutResource(): Int = R.layout.activity_main

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onResume() {
        super.onResume()
        presenter.onSplashAction()
    }

    override fun navigate(navigationType: SplashContract.NavigationType?) {
        when (navigationType) {
            SplashContract.NavigationType.HOME -> openHomeScreen()
            SplashContract.NavigationType.LOGIN -> openLoginScreen()
        }
    }

    private fun openHomeScreen() = startActivity(HomeActivity::class.java)

    private fun openLoginScreen() = startActivity(LoginActivity::class.java)


}
